src/exporter/DaemonMetricCollector.cc

```diff
@@ -107,16 +129,91 @@ void DaemonMetricCollector::dump_asok_metrics() {
+      "Time spent scraping and transforming perfcounters to metrics");
+  std::string scrap_labels("host=");
+  std::string host = ceph_get_hostname();
+  scrap_labels += quote(host);
+  add_metric(ss, timer.get_ms(), "ceph_exporter_scrape_time", scrap_desc,
+             "gauge", scrap_labels);
+
   const std::lock_guard<std::mutex> lock(metrics_mutex);
   metrics = ss.str();
+  metrics += get_process_metrics(daemon_pids);
+}
+
+std::vector<std::string> read_proc_stat_file(std::string path) {
+  std::string stat = read_file_to_string(path);
+  std::vector<std::string> strings;
+  auto parts = ceph::split(stat);
+  strings.assign(parts.begin(), parts.end());
+  return strings;
+}
+
+struct pstat read_pid_stat(int pid) {
+  std::string stat_path("/proc/" + std::to_string(pid) + "/stat");
+  std::vector<std::string> stats = read_proc_stat_file(stat_path);
+  struct pstat stat;
+  stat.minflt = std::stoul(stats[9]);
+  stat.majflt = std::stoul(stats[11]);
+  stat.utime = std::stoul(stats[13]);
+  stat.stime = std::stoul(stats[14]);
+  stat.num_threads = std::stoul(stats[19]);
+  stat.start_time = std::stoul(stats[21]);
+  stat.vm_size = std::stoul(stats[22]);
+  stat.resident_size = std::stoi(stats[23]);
+  return stat;
+}
+
+std::string DaemonMetricCollector::get_process_metrics(
+    std::vector<std::pair<std::string, int>> daemon_pids) {
+  std::string path("/proc");
+  std::stringstream ss;
+  for (auto &[daemon_name, pid] : daemon_pids) {
+    std::vector<std::string> uptimes = read_proc_stat_file("/proc/uptime");
+    struct pstat stat = read_pid_stat(pid);
+    int clk_tck = sysconf(_SC_CLK_TCK);
+    double start_time_seconds = stat.start_time / (double)clk_tck;
+		double user_time = stat.utime / (double)clk_tck;
+		double kernel_time = stat.stime / (double)clk_tck;
+    double total_time_seconds = user_time + kernel_time;
+    double uptime = std::stod(uptimes[0]);
+    double elapsed_time = uptime - start_time_seconds;
+		double idle_time = elapsed_time  - total_time_seconds;
+    double usage = total_time_seconds * 100 / elapsed_time;
+
+    std::string labels = "ceph_daemon=";
+    labels += quote(daemon_name);
+    add_metric(ss, stat.minflt, "ceph_exporter_minflt",
+               "Number of minor page faults of daemon", "gauge", labels);
+    add_metric(ss, stat.majflt, "ceph_exporter_majflt",
+               "Number of major page faults of daemon", "gauge", labels);
+    add_metric(ss, stat.num_threads, "ceph_exporter_num_threads",
+               "Number of threads used by daemon", "gauge", labels);
+    add_metric(ss, usage, "ceph_exporter_cpu_usage", "CPU usage of a daemon",
+               "gauge", labels);
+    add_metric(ss, kernel_time, "ceph_exporter_cpu_kernel", "Process time in kernel mode",
+               "gauge", labels);
+    add_metric(ss, user_time, "ceph_exporter_cpu_user", "Process time in user mode",
+               "gauge", labels);
+    add_metric(ss, idle_time, "ceph_exporter_cpu_user", "Process idle time",
```

Comment:
You used the same name it should probably be `ceph_exporter_cpu_idle` I guess. I think it would be better to align on node-exporter and have the same metric with a label "mode" (so that would be mode="system" or mode="user" or mode="idle" here probably).